src/components/TimePicker.vue

```diff
@@ -0,0 +1,197 @@
+    </v-row>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator';
+import SwiperClass, { SwiperOptions } from 'swiper';
+import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
+
+@Component({
+    components: {
+        Swiper,
+        SwiperSlide,
+    },
+})
+export default class AttractionTime extends Vue {
+    @Prop() currentDate?: Date;
+
+    private hours = [
+        '01',
+        '02',
+        '03',
+        '04',
+        '05',
+        '06',
+        '07',
+        '08',
+        '09',
+        '10',
+        '11',
+        '12',
+        '13',
+        '14',
+        '15',
+        '16',
+        '17',
+        '18',
+        '19',
+        '20',
+        '21',
+        '22',
+        '23',
+        '00',
+    ];
+
+    private minutes = [
+        '00',
+        '05',
+        '10',
+        '15',
+        '20',
+        '25',
+        '30',
+        '35',
+        '40',
+        '45',
+        '50',
+        '55',
+    ];
+
+    data(): {
+        activeHour: string;
+        activeMinute: string;
+        swiperOption: SwiperOptions;
+    } {
+        return {
+            activeHour: '08',
+            activeMinute: '15',
+            swiperOption: {
+                direction: 'vertical',
+                effect: 'coverflow',
+                coverflowEffect: {
+                    rotate: 0,
+                    stretch: 0,
+                    depth: 100,
+                    modifier: 1,
+                    slideShadows: false,
+                },
+                centeredSlides: true,
+                slidesPerView: 6,
+                slideToClickedSlide: true,
+            },
+        };
+    }
+
+    created(): void {
+        this.$data.activeHour = this.$props.currentDate.getHours().toString();
+        this.$data.activeMinute = this.$props.currentDate
+            .getMinutes()
+            .toString();
+    }
+
+    updateCurrentTime(): void {
+        const newDate = new Date(this.$props.currentDate);
+        newDate.setHours(Number(this.$data.activeHour));
+        newDate.setMinutes(Number(this.$data.activeMinute));
+        this.$emit('update:currentDate', newDate);
+    }
+
+    get currentHourIndex(): number {
+        return this.hours.findIndex((hour) => +hour == +this.$data.activeHour);
+    }
+
+    get currentMinuteIndex(): number {
+        return this.minutes.findIndex(
+            (minute) => +minute == +this.$data.activeMinute,
+        );
+    }
+
+    mounted(): void {
+        this.$nextTick(() => {
+            (this.$refs.hourSwiper as HTMLFormElement)?.$swiper.slideTo(
+                this.currentHourIndex,
+            );
+
+            (this.$refs.minuteSwiper as HTMLFormElement)?.$swiper.slideTo(
+                this.currentMinuteIndex,
+            );
+
+            const updateActiveHour = (hourIndex: number) => {
+                this.$data.activeHour = this.$data.hours[hourIndex];
+                this.updateCurrentTime();
```

Comment:
send the this.$data.hours[hourIndex]; to the update function instead of saving the value